fmriprep/workflows/fieldmap/unwarp.py

```diff
@@ -215,10 +221,268 @@ def init_sdc_unwarp_wf(reportlets_dir, ants_nthreads, fmap_bspline,
+    
+    This procedure works if there is only one '_epi' file is present
+    (as long as it has the opposite phase encoding direction to the target 
+    file). The target file will be used to estimate the field distorton.
+    However, if there is another '_epi' file present with a matching
+    phase encoding direction to the target it will be used instead.
+    
+    Currently different phase encoding dimension between target file and the 
+    '_epi' file(s) (for example 'i' and 'j') is not supported.
+    
+    The warp field correcting for the distortions is estimated using AFNI's 
+    3dQwarp with displacement estimation limited to the target file phase
+    encoding direction.
+
+    It also calculates a new mask for the input dataset that takes into 
+    account the distortions.
+
+    .. workflow ::
+
+        from fmriprep.workflows.fieldmap.unwarp import init_pepolar_unwarp_wf
+        wf = init_pepolar_unwarp_wf(fmaps=['/dataset/sub-01/fmap/sub-01_epi.nii.gz'],
+                                    fmaps_pes=['j-'],
+                                    bold_file='/dataset/sub-01/func/sub-01_task-rest_bold.nii.gz',
+                                    bold_file_pe='j',
+                                    ants_nthreads=8)
+
+
+    Inputs
+
+        in_reference
+            the reference image
+        in_mask
+            a brain mask corresponding to ``in_reference``
+        name_source
+            not used, kept for signature compatibility with ``init_sdc_unwarp_wf``
+
+    Outputs
+
+        out_reference
+            the ``in_reference`` after unwarping
+        out_warp
+            the corresponding :abbr:`DFM (displacements field map)` compatible with
+            ANTs
+        out_mask
+            mask of the unwarped input file
+        out_mask_report
+            reportled for the skullstripping
+
+    """
+    if not bold_file_pe:
+        bold_file_pe = layout.get_metadata(bold_file)["PhaseEncodingDirection"]
+
+    usable_fieldmaps_matching_pe = []
+    usable_fieldmaps_opposite_pe = []
+    args = '-noXdis -noYdis -noZdis'
+    rm_arg = {'i': '-noXdis',
+              'j': '-noYdis',
+              'k': '-noZdis'}[bold_file_pe[0]]
+    args = args.replace(rm_arg, '')
+
+    for i, fmap in enumerate(fmaps):
+        if fmaps_pes:
+            fmap_pe = fmaps_pes[i]
+        else:
+            fmap_pe = layout.get_metadata(fmap)["PhaseEncodingDirection"]
+        if fmap_pe[0] == bold_file_pe[0]:
+            if len(fmap_pe) != len(bold_file_pe):
+                add_list = usable_fieldmaps_opposite_pe
+            else:
+                add_list = usable_fieldmaps_matching_pe
+            add_list.append(fmap)
+
+    if len(usable_fieldmaps_opposite_pe) == 0:
+        raise Exception("None of the discovered fieldmaps has the right "
+                        "phase encoding direction. Possibly a problem with"
+                        "metadata. If not rerun with '--ignore fieldmaps' to"
```

Comment:
Comma after "If not". Add a space at the end of each of the preceding two strings.